### PR TITLE
[8.0] purchase internal validation

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: purchase_internal_validation/README.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
+============================
```
Comment (elicoidal) [inline purchase_internal_validation/README.rst:1]: Hi @dufresnedavid Thanks for the contribution.
I miss the badge at beginning of README

File: purchase_internal_validation/README.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+Usage
+=====
+
+* As a Purchase user, go to Purchase > Purchases > Quotations
+* Create a quotation exceeding the limit and try to confirm it
+* As a Purchase validator, validate or refuse the quotation
+* As the user, you can now confirm the PO
+
+For further information, please visit:
```
Comment (elicoidal) [inline purchase_internal_validation/README.rst:28]: remove this part

File: purchase_internal_validation/README.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+* https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
```
Comment (elicoidal) [inline purchase_internal_validation/README.rst:32]: remove if empty

File: purchase_internal_validation/README.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+Bug Tracker
+===========
+
+Bugs are tracked on
+`GitHub Issues <https://github.com/OCA/purchase-workflow/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and
+welcomed feedback
+`here <https://github.com/OCA/purchase-workflow/issues/new>`_.
```
Comment (elicoidal) [inline purchase_internal_validation/README.rst:45]: This is not the standard bug tracker information. Please check template (https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst)

File: purchase_internal_validation/README.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline purchase_internal_validation/README.rst:71]: https

File: purchase_internal_validation/__init__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline purchase_internal_validation/__init__.py:1]: - Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: purchase_internal_validation/models/purchase_order.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    STATE_SELECTION = [
+        ('draft', 'Draft PO'),
+        ('sent', 'RFQ'),
+        ('bid', 'Bid Received'),
+
+        # States Added
+        ('wait_valid', 'Waiting for Validation'),
```
Comment (elicoidal) [inline purchase_internal_validation/models/purchase_order.py:38]: a description of the states in README would be appreciated

File: purchase_internal_validation/models/purchase_order.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        ('wait_correct', 'Waiting for Correction'),
+        ('wait', 'Waiting'),
```
Comment (elicoidal) [inline purchase_internal_validation/models/purchase_order.py:40]: Looks like a lot of steps. What is Waiting used for?

File: purchase_internal_validation/models/purchase_order.py [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        return True
+
+    @api.multi
+    def test_require_validation(self):
+        limit = (
+            self.env["purchase.config.settings"].get_default_limit_amount()
+            ["limit_amount"]
+        )
+        for rec in self:
+            if rec.amount_total >= limit:
```
Comment (elicoidal) [inline purchase_internal_validation/models/purchase_order.py:85]: Main problem I see in this module (same as the official one actually) is that no currency is provided.
When you deal with multiple currencies this behavior makes no sense or even is dangerous.
At least you should convert the amount_total in company currency before comparing to the limit.

**Comment by ghost** [comment]
@elicoidal thanks for the review and sorry for responding late. I agree with your point.

This module is not generic. It misses multi-company and currency support. For now, it meets the requirements of my client, so I will not invest more time on it. For that reason, I am closing this pull request.

**Comment by elicoidal** [comment]
@dufresnedavid Thanks forthe feedback: next time we will have better luck!

**Pull request closed without merging**